backend/service/k8s/pods.go

```diff
@@ -62,6 +62,79 @@ func (s *svc) ListPods(ctx context.Context, clientset, cluster, namespace string
 	return pods, nil
 }
 
+func (s *svc) UpdatePodAnnotations(ctx context.Context, clientset, cluster, namespace, name string, expectedAnnotations, newAnnotations Annotations) error {
+	cs, err := s.manager.GetK8sClientset(clientset, cluster, namespace)
+	if err != nil {
+		return err
+	}
+
+	pod, err := cs.CoreV1().Pods(namespace).Get(name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	podAnnotations := pod.GetAnnotations()
+
+	err = checkPreconditions(expectedAnnotations, podAnnotations)
+	if err != nil {
+		return err
+	}
+
+	for newAnnotations, newValue := range newAnnotations {
+		if newValue != nil {
+			podAnnotations[newAnnotations] = newValue.GetValue()
+		} else {
+			delete(podAnnotations, newAnnotations)
+		}
+	}
```

Comment:
This is very destructive and should avoid deleting annotations that we are not controlling via clutch. For example at Lyft there are several k8s systems that rely on pod annotations such as secret injection (confidant).

Reply:
ah nvm this is fine